<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VSPSerialIO</class>
 <widget class="QDialog" name="VSPSerialIO">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>417</width>
    <height>694</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>VSP Serial I/O</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../vspui.qrc">
    <normaloff>:/assets/png/vspclient_7.png</normaloff>:/assets/png/vspclient_7.png</iconset>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>24</number>
   </property>
   <property name="leftMargin">
    <number>8</number>
   </property>
   <property name="topMargin">
    <number>8</number>
   </property>
   <property name="rightMargin">
    <number>8</number>
   </property>
   <property name="bottomMargin">
    <number>8</number>
   </property>
   <item>
    <widget class="QGroupBox" name="gbxSerialPort">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Serial Port</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="spacing">
       <number>12</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_25">
        <property name="text">
         <string>Serial Port</string>
        </property>
        <property name="buddy">
         <cstring>cbxComPort</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_26">
        <property name="text">
         <string>Baud Rate:</string>
        </property>
        <property name="buddy">
         <cstring>cbxBaud</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="cbxBaud">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_28">
        <property name="text">
         <string>Stop Bits:</string>
        </property>
        <property name="buddy">
         <cstring>cbxStopBits</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="cbxStopBits">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_30">
        <property name="text">
         <string>Parity:</string>
        </property>
        <property name="buddy">
         <cstring>cbxParity</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="cbxParity">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_27">
        <property name="text">
         <string>Data Bits:</string>
        </property>
        <property name="buddy">
         <cstring>cbxDataBits</cstring>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="cbxDataBits">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="4">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>12</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btnConnect">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="cbxDtr">
        <property name="text">
         <string>DTR</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QCheckBox" name="cbxRts">
        <property name="text">
         <string>RTS</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QCheckBox" name="cbxCts">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>CTS</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QCheckBox" name="cbxDSR">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>DSR</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_29">
        <property name="text">
         <string>Flow Ctrl.:</string>
        </property>
        <property name="buddy">
         <cstring>cbxFlowControl</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QComboBox" name="cbxFlowControl">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="3">
       <widget class="QComboBox" name="cbxComPort">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>15</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gbxOutput">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Output</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="spacing">
       <number>12</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Send line:</string>
        </property>
        <property name="buddy">
         <cstring>edOutputLine</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Looper length:</string>
        </property>
        <property name="buddy">
         <cstring>edGenLength</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="cbxLineEnding">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>CR/LF</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>LF</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CR</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>$</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>|</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>:</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>;</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Line ending:</string>
        </property>
        <property name="buddy">
         <cstring>cbxLineEnding</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="btnLooper">
        <property name="text">
         <string>Looper</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QSpinBox" name="edGenLength">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>16384</number>
        </property>
        <property name="value">
         <number>16</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QLineEdit" name="edOutputLine">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnSendLine">
        <property name="text">
         <string>Send Line</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="btnSendFile">
        <property name="text">
         <string>Send file...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gbxInput">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Input</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>4</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <widget class="QPlainTextEdit" name="txInputView">
        <property name="tabChangesFocus">
         <bool>true</bool>
        </property>
        <property name="documentTitle">
         <string>Input</string>
        </property>
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::LineWrapMode::NoWrap</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::TextInteractionFlag::TextSelectableByKeyboard|Qt::TextInteractionFlag::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="txOutputInfo">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <action name="actionClose">
   <property name="icon">
    <iconset theme="process-stop"/>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="toolTip">
    <string>Close window</string>
   </property>
  </action>
  <action name="actionNewWindow">
   <property name="icon">
    <iconset theme="document-open"/>
   </property>
   <property name="text">
    <string>New...</string>
   </property>
   <property name="toolTip">
    <string>Open new window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>cbxComPort</tabstop>
  <tabstop>cbxBaud</tabstop>
  <tabstop>cbxStopBits</tabstop>
  <tabstop>cbxParity</tabstop>
  <tabstop>cbxDataBits</tabstop>
  <tabstop>cbxFlowControl</tabstop>
  <tabstop>cbxDtr</tabstop>
  <tabstop>cbxRts</tabstop>
  <tabstop>cbxCts</tabstop>
  <tabstop>cbxDSR</tabstop>
  <tabstop>btnConnect</tabstop>
  <tabstop>edOutputLine</tabstop>
  <tabstop>edGenLength</tabstop>
  <tabstop>cbxLineEnding</tabstop>
  <tabstop>btnLooper</tabstop>
  <tabstop>txInputView</tabstop>
 </tabstops>
 <resources>
  <include location="../vspui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>VSPSerialIO</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
